packages/explat-client/src/internal/test/state.ts

```diff
@@ -6,48 +6,37 @@ import { validExperimentAssignment, validFallbackExperimentAssignment } from '..
 
 describe( 'state', () => {
 	it( 'should save and retrieve valid ExperimentAssignments', () => {
-		expect( State.retrieveExperimentAssignment( validExperimentAssignment.experimentName ) ).toBe(
-			undefined
-		);
-		State.storeExperimentAssignment( validExperimentAssignment );
+		const store = State.createStore();
+		expect(
+			State.retrieveExperimentAssignment( store, validExperimentAssignment.experimentName )
+		).toBe( undefined );
+		State.storeExperimentAssignment( store, validExperimentAssignment );
 		expect(
-			State.retrieveExperimentAssignment( validExperimentAssignment.experimentName )
+			State.retrieveExperimentAssignment( store, validExperimentAssignment.experimentName )
 		).toEqual( validExperimentAssignment );
 
 		expect(
-			State.retrieveExperimentAssignment( validFallbackExperimentAssignment.experimentName )
+			State.retrieveExperimentAssignment( store, validFallbackExperimentAssignment.experimentName )
 		).toBe( undefined );
-		State.storeExperimentAssignment( validFallbackExperimentAssignment );
+		State.storeExperimentAssignment( store, validFallbackExperimentAssignment );
 		expect(
-			State.retrieveExperimentAssignment( validFallbackExperimentAssignment.experimentName )
+			State.retrieveExperimentAssignment( store, validFallbackExperimentAssignment.experimentName )
 		).toEqual( validFallbackExperimentAssignment );
 	} );
 
 	it( 'should throw for storing an ExperimentAssignment for a currently stored Experiment with an older date', () => {
+		const store = State.createStore();
+		State.storeExperimentAssignment( store, validFallbackExperimentAssignment );
```

Comment:
I'm not sure what benefits we get from passing in `store` everywhere and we probably don't want experimenters just storing any random variation, bypassing the API. It smells like it might cause an issue if someone deploys that by accident, or even worse, they deploy it and it only gets called under certain conditions, thereby breaking their experiment.

Reply:
The store is local to a ExPlatClient and can't be accessed from outside of it.
These functions are also within a `src/internal` and not meant to be used outside of it.

(This touches on the Class vs Function argument btw)

Reply:
> This touches on the Class vs Function argument btw

Yeah, but arguing over implementation details/preferences is not worth it IMHO. I just want to understand the need for this abstraction when it didn't seem to change much in the tests. If it needs to exist or makes things better, then we should keep it.

> The store is local to a ExPlatClient and can't be accessed from outside of it.
> These functions are also within a src/internal and not meant to be used outside of it.

I'm still wary about it. People will use whatever is necessary to accomplish a task, especially if there's a time crunch and/or they don't understand the implications.

Reply:
> If it needs to exist or makes things better, then we should keep it.

Oh you can see it in action in the next PR: https://github.com/Automattic/wp-calypso/pull/48475/files#diff-6ccc8c485f7dc1c524767b67ae2efa00edf70f04ab35c903a9f0287b644d507aR63

Reply:
> Oh you can see it in action in the next PR

Ok, here's my feedback getting a better picture of how this is all laid out.

1. The `state` object passed to the client is a leaky-abstraction. The client should never (be allowed to) manipulate the state directly if we're going to encapsulate it here.
2. From a client's perspective, the `State` object is more like a key to access other functions than something useful. Perhaps we should get rid of the `state` object we return, and instead, just use `Symbol.for( experimentName )` as the key.
3. Each time we create a client, it appears to be an entirely new client with new state. This has some interesting implications if not wielded correctly. I could also be reading it wrong. We should have some stable-state, perhaps keyed by Symbol, so we don't have nested components each creating their own clients and doing their own API calls.

Reply:
> The state object passed to the client is a leaky-abstraction. The client should never (be allowed to) manipulate the state directly if we're going to encapsulate it here.

Ah, good point about encapsulation being necessary here, this does tip the balance into class territory for me.

> Each time we create a client, it appears to be an entirely new client with new state. This has some interesting implications if not wielded correctly. I could also be reading it wrong. We should have some stable-state, perhaps keyed by Symbol, so we don't have nested components each creating their own clients and doing their own API calls.

Since we are doing single fetch experiments, the only thing that I can see go wrong is a single experiment using two separate libs, with slightly different timing giving half implemented variations. I think when we get more clients within the same codebase we can add in a dev-mode error for multiple clients